In [1]:
import pandas as pd

In [3]:
# Read the Excel file into a Pandas DataFrame
airbnb = pd.read_excel('airbnb.xlsx')

In [4]:
airbnb.head()

Host Id Host Since                                Name Neighbourhood    
0   5162530        NaT     1 Bedroom in Prime Williamsburg       Brooklyn  \
1  33134899        NaT     Sunny, Private room in Bushwick       Brooklyn   
2  39608626        NaT                Sunny Room in Harlem      Manhattan   
3       500 2008-06-26  Gorgeous 1 BR with Private Balcony      Manhattan   
4       500 2008-06-26            Trendy Times Square Loft      Manhattan   

  Property Type  Review Scores Rating (bin)        Room Type  Zipcode  Beds   
0     Apartment                         NaN  Entire home/apt  11249.0   1.0  \
1     Apartment                         NaN     Private room  11206.0   1.0   
2     Apartment                         NaN     Private room  10032.0   1.0   
3     Apartment                         NaN  Entire home/apt  10024.0   3.0   
4     Apartment                        95.0     Private room  10036.0   3.0   

   Number of Records  Number Of Reviews  Price  Review Scores Rating  
0                  1                  0    145                   NaN  
1                  1                  1     37                   NaN  
2                  1                  1     28                   NaN  
3                  1                  0    199                   NaN  
4                  1                 39    549                  96.0

In [5]:
# descriptive of available data#
airbnb.describe()

Host Id                     Host Since   
count  3.047800e+04                          30475  \
mean   1.273171e+07  2013-06-29 17:56:15.317473280   
min    5.000000e+02            2008-06-26 00:00:00   
25%    2.701298e+06            2012-06-21 00:00:00   
50%    8.551693e+06            2013-09-01 00:00:00   
75%    2.120617e+07            2014-09-12 00:00:00   
max    4.303307e+07            2015-08-31 00:00:00   
std    1.190270e+07                            NaN   

       Review Scores Rating (bin)       Zipcode          Beds   
count                22155.000000  30344.000000  30393.000000  \
mean                    90.738659  10584.854831      1.530089   
min                     20.000000   1003.000000      0.000000   
25%                     85.000000  10017.000000      1.000000   
50%                     90.000000  10065.000000      1.000000   
75%                    100.000000  11216.000000      2.000000   
max                    100.000000  99135.000000     16.000000   
std                      9.059519    921.299397      1.015359   

       Number of Records  Number Of Reviews         Price   
count            30478.0       30478.000000  30478.000000  \
mean                 1.0          12.018735    163.589737   
min                  1.0           0.000000     10.000000   
25%                  1.0           0.000000     80.000000   
50%                  1.0           3.000000    125.000000   
75%                  1.0          13.000000    195.000000   
max                  1.0         257.000000  10000.000000   
std                  0.0          21.980703    197.785454   

       Review Scores Rating  
count          22155.000000  
mean              91.993230  
min               20.000000  
25%               89.000000  
50%               94.000000  
75%              100.000000  
max              100.000000  
std                8.850373

In [6]:
# More information on data quality

airbnb.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 30478 entries, 0 to 30477
Data columns (total 13 columns):
 #   Column                      Non-Null Count  Dtype         
---  ------                      --------------  -----         
 0   Host Id                     30478 non-null  int64         
 1   Host Since                  30475 non-null  datetime64[ns]
 2   Name                        30478 non-null  object        
 3   Neighbourhood               30478 non-null  object        
 4   Property Type               30475 non-null  object        
 5   Review Scores Rating (bin)  22155 non-null  float64       
 6   Room Type                   30478 non-null  object        
 7   Zipcode                     30344 non-null  float64       
 8   Beds                        30393 non-null  float64       
 9   Number of Records           30478 non-null  int64         
 10  Number Of Reviews           30478 non-null  int64         
 11  Price                       30478 non-null  int64     

In [7]:
# Let's remove duplicate first before working on the null values#

airbnb = airbnb.drop_duplicates() 

Number of records went down from 30478 to 30461

In [8]:
airbnb.info()

<class 'pandas.core.frame.DataFrame'>
Index: 30461 entries, 0 to 30477
Data columns (total 13 columns):
 #   Column                      Non-Null Count  Dtype         
---  ------                      --------------  -----         
 0   Host Id                     30461 non-null  int64         
 1   Host Since                  30458 non-null  datetime64[ns]
 2   Name                        30461 non-null  object        
 3   Neighbourhood               30461 non-null  object        
 4   Property Type               30458 non-null  object        
 5   Review Scores Rating (bin)  22152 non-null  float64       
 6   Room Type                   30461 non-null  object        
 7   Zipcode                     30327 non-null  float64       
 8   Beds                        30376 non-null  float64       
 9   Number of Records           30461 non-null  int64         
 10  Number Of Reviews           30461 non-null  int64         
 11  Price                       30461 non-null  int64         


Decision about Non null values
Host Since: I have 3 records without date.I decided to remove them as a missing a date will affect any time serie analysis
Property type: I will remove them because I will be using property type for comparison analysis
Review rating: I decided to keep null records because they represent 1/3 of my data. I will exclude them during the rating analysis
Zipcode: I decided to remove record without zip codes as I am planing to Map my locations
beds: It will great to have beds for comarison analysisb

In [10]:
airbnb.columns

Index(['Host Id', 'Host Since', 'Name', 'Neighbourhood ', 'Property Type',
       'Review Scores Rating (bin)', 'Room Type', 'Zipcode', 'Beds',
       'Number of Records', 'Number Of Reviews', 'Price',
       'Review Scores Rating'],
      dtype='object')

In [12]:
# removing null values

airbnb = airbnb.dropna(subset=['Host Since', 'Property Type','Zipcode','Beds'])

In [16]:
airbnb.info()

<class 'pandas.core.frame.DataFrame'>
Index: 30237 entries, 3 to 30477
Data columns (total 13 columns):
 #   Column                      Non-Null Count  Dtype         
---  ------                      --------------  -----         
 0   Host Id                     30237 non-null  int64         
 1   Host Since                  30237 non-null  datetime64[ns]
 2   Name                        30237 non-null  object        
 3   Neighbourhood               30237 non-null  object        
 4   Property Type               30237 non-null  object        
 5   Review Scores Rating (bin)  22007 non-null  float64       
 6   Room Type                   30237 non-null  object        
 7   Zipcode                     30237 non-null  float64       
 8   Beds                        30237 non-null  float64       
 9   Number of Records           30237 non-null  int64         
 10  Number Of Reviews           30237 non-null  int64         
 11  Price                       30237 non-null  int64         


In [15]:
airbnb.isnull().sum()

Host Id                          0
Host Since                       0
Name                             0
Neighbourhood                    0
Property Type                    0
Review Scores Rating (bin)    8230
Room Type                        0
Zipcode                          0
Beds                             0
Number of Records                0
Number Of Reviews                0
Price                            0
Review Scores Rating          8230
dtype: int64

In [17]:
airbnb.describe()

Host Id                     Host Since   
count  3.023700e+04                          30237  \
mean   1.274262e+07  2013-06-30 10:50:06.726857984   
min    5.000000e+02            2008-06-26 00:00:00   
25%    2.702080e+06            2012-06-21 00:00:00   
50%    8.571843e+06            2013-09-03 00:00:00   
75%    2.121866e+07            2014-09-13 00:00:00   
max    4.303307e+07            2015-08-31 00:00:00   
std    1.190319e+07                            NaN   

       Review Scores Rating (bin)       Zipcode          Beds   
count                22007.000000  30237.000000  30237.000000  \
mean                    90.732949  10584.541522      1.529351   
min                     20.000000   1003.000000      0.000000   
25%                     85.000000  10017.000000      1.000000   
50%                     90.000000  10065.000000      1.000000   
75%                    100.000000  11216.000000      2.000000   
max                    100.000000  99135.000000     16.000000   
std                      9.069640    922.221036      1.015054   

       Number of Records  Number Of Reviews         Price   
count            30237.0       30237.000000  30237.000000  \
mean                 1.0          12.001455    163.560505   
min                  1.0           0.000000     10.000000   
25%                  1.0           0.000000     81.000000   
50%                  1.0           3.000000    125.000000   
75%                  1.0          13.000000    195.000000   
max                  1.0         257.000000  10000.000000   
std                  0.0          21.921968    197.392068   

       Review Scores Rating  
count          22007.000000  
mean              91.987731  
min               20.000000  
25%               89.000000  
50%               94.000000  
75%              100.000000  
max              100.000000  
std                8.860796

Based on Info and describe, we have the following:
8230 records without rating
25% of data has no reviews (number of reviews)
Some records have a price of 10000 - To verify that it is not a typob

In [35]:
# verifying the records with a price 10000



airbnb.groupby('Price')['Price'].agg(['count']).sort_values('Price', ascending=False) .head(20)


count
Price       
10000      1
8000       1
6500       1
5999       1
5000       4
4750       1
4500       2
4000       3
3750       1
3500       1
3390       1
3200       1
3100       1
3000       4
2750       2
2695       1
2599       1
2520       1
2500       9
2499       1

Well It seems that the 10000 is not a typo and there are many other locations with high price.

In [37]:
airbnb.sort_values(by="Price", ascending=False).head(20)


Host Id Host Since                                 Name   
23927  23248648 2014-11-02       NO LONGER BOOKING RESERVATIONS  \
4002    1177497 2011-09-18                        Film Location   
17900  11598359 2014-01-24     SUPER BOWL Brooklyn Duplex Apt!!   
2405     496780 2011-04-09       test listing which is not real   
15633   9056494 2013-09-25      Super football SB weekend deal.   
1275     213266 2010-08-26   Beautiful 1 Bedroom in Nolita/Soho   
24136  23564279 2014-11-10  Breathtaking 2-Story Soho Penthouse   
7550    2675644 2012-06-18    Turn of the Century Film Location   
3483     904192 2011-08-01   SOHO PENTHOUSE W/ PRIVATE ROOFDECK   
17466  11026037 2014-01-04   Beautiful 2br/2bath 1300 sq ft apt   
4003    1177497 2011-09-18                 Photography Location   
17429  10967533 2014-01-02   LES 3BR Duplex w/ Private Backyard   
10237   4204562 2012-11-20       MOVIE SHOOTS! 14,000sf MANSION   
17335  10854661 2013-12-27                  Extraordinary Space   
4781    1427243 2011-11-20    Prime NYC Location for Super Bowl   
27088  31433828 2015-04-18  1.5 month lease near Lincoln Center   
28951  37155810 2015-06-30         Central Park Luxury : 115916   
17482  11052635 2014-01-05            L BIG STUDIO WITH TERRACE   
28733  36538690 2015-06-23         Park Slope Brooklyn for July   
15181   8503180 2013-08-28               East Village Townhouse   

      Neighbourhood  Property Type  Review Scores Rating (bin)   
23927      Manhattan     Apartment                        95.0  \
4002        Brooklyn         House                         NaN   
17900       Brooklyn     Apartment                         NaN   
2405        Brooklyn     Apartment                         NaN   
15633      Manhattan     Apartment                         NaN   
1275       Manhattan     Apartment                        75.0   
24136      Manhattan     Apartment                         NaN   
7550   Staten Island         House                        40.0   
3483       Manhattan     Apartment                         NaN   
17466      Manhattan     Apartment                         NaN   
4003        Brooklyn         House                       100.0   
17429      Manhattan     Apartment                         NaN   
10237          Bronx         House                         NaN   
17335         Queens         Other                         NaN   
4781       Manhattan     Apartment                         NaN   
27088      Manhattan     Apartment                       100.0   
28951      Manhattan     Apartment                         NaN   
17482      Manhattan     Apartment                         NaN   
28733       Brooklyn     Apartment                       100.0   
15181      Manhattan         House                       100.0   

             Room Type  Zipcode  Beds  Number of Records  Number Of Reviews   
23927  Entire home/apt  10025.0   2.0                  1                 21  \
4002   Entire home/apt  11205.0   5.0                  1                  0   
17900  Entire home/apt  11238.0   4.0                  1                  0   
2405   Entire home/apt  11249.0   5.0                  1                  0   
15633  Entire home/apt  10040.0   2.0                  1                  0   
1275   Entire home/apt  10012.0   2.0                  1                  2   
24136  Entire home/apt  10013.0   3.0                  1                  0   
7550   Entire home/apt  10301.0  12.0                  1                  1   
3483   Entire home/apt  10012.0   3.0                  1                  0   
17466  Entire home/apt  10022.0   2.0                  1                  0   
4003   Entire home/apt  11205.0   5.0                  1                  3   
17429  Entire home/apt  10002.0   3.0                  1                  0   
10237  Entire home/apt  10471.0   8.0                  1                  0   
17335  Entire home/apt  11101.0   1.0                  1                  0   
4781   Entire home/apt  1000

In [ ]:
airbnb = airbnb.drop('column_name', axis=1)

Looking further Into the details for the high priced locations many of them have no reviews and the one with 10000 is not receving any more bookings. We can remove it from the data set if we want. 

In [39]:
airbnb = airbnb.loc[airbnb['Price'] != 10000]

In [40]:
airbnb.describe()

Host Id                     Host Since   
count  3.023600e+04                          30236  \
mean   1.274227e+07  2013-06-30 10:26:47.831723776   
min    5.000000e+02            2008-06-26 00:00:00   
25%    2.702063e+06            2012-06-21 00:00:00   
50%    8.570398e+06            2013-09-03 00:00:00   
75%    2.121861e+07            2014-09-13 00:00:00   
max    4.303307e+07            2015-08-31 00:00:00   
std    1.190324e+07                            NaN   

       Review Scores Rating (bin)       Zipcode          Beds   
count                22006.000000  30236.000000  30236.000000  \
mean                    90.732755  10584.560028      1.529336   
min                     20.000000   1003.000000      0.000000   
25%                     85.000000  10017.000000      1.000000   
50%                     90.000000  10065.000000      1.000000   
75%                    100.000000  11216.000000      2.000000   
max                    100.000000  99135.000000     16.000000   
std                      9.069800    922.230673      1.015067   

       Number of Records  Number Of Reviews         Price   
count            30236.0       30236.000000  30236.000000  \
mean                 1.0          12.001158    163.235183   
min                  1.0           0.000000     10.000000   
25%                  1.0           0.000000     81.000000   
50%                  1.0           3.000000    125.000000   
75%                  1.0          13.000000    195.000000   
max                  1.0         257.000000   8000.000000   
std                  0.0          21.922270    189.115558   

       Review Scores Rating  
count          22006.000000  
mean              91.987594  
min               20.000000  
25%               89.000000  
50%               94.000000  
75%              100.000000  
max              100.000000  
std                8.860974

In [41]:
airbnb=airbnb.drop('Number of Records', axis=1)

In [42]:
airbnb.head()

Host Id Host Since                                 Name Neighbourhood    
3      500 2008-06-26   Gorgeous 1 BR with Private Balcony      Manhattan  \
4      500 2008-06-26             Trendy Times Square Loft      Manhattan   
5     1039 2008-07-25   Big Greenpoint 1BD w/ Skyline View       Brooklyn   
6     1783 2008-08-12                         Amazing Also      Manhattan   
7     2078 2008-08-15  Colorful, quiet, & near the subway!       Brooklyn   

  Property Type  Review Scores Rating (bin)        Room Type  Zipcode  Beds   
3     Apartment                         NaN  Entire home/apt  10024.0   3.0  \
4     Apartment                        95.0     Private room  10036.0   3.0   
5     Apartment                       100.0  Entire home/apt  11222.0   1.0   
6     Apartment                       100.0  Entire home/apt  10004.0   1.0   
7     Apartment                        90.0     Private room  11201.0   1.0   

   Number Of Reviews  Price  Review Scores Rating  
3                  0    199                   NaN  
4                 39    549                  96.0  
5                  4    149                 100.0  
6                  9    250                 100.0  
7                 80     90                  94.0

In [44]:
airbnb.to_excel('airbnb_clean.xlsx', index=False)